### PR TITLE
Add ToPureTensor transform

### PR DESCRIPTION
**Pull request by NicolasHug**

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -2270,3 +2270,24 @@ def test_image_correctness(self, permutation, batch_dims):
...
+    def test_correctness(self):
+        input = {
+            "img": make_image(),
+            "img_tensor": make_image_tensor(),
+            "img_pil": make_image_pil(),
+            "mask": make_detection_mask(),
+            "video": make_video(),
+            "bbox": make_bounding_box(),
+            "str": "str",
+        }
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:2285]: Can we parametrize over that rather than looping below?

**Review by pmeier — APPROVED** [review]
Some comments inline. Nothing major so stamping to unblock.

File: torchvision/transforms/v2/_type_conversion.py [hunk 2]
```diff
@@ -80,3 +80,17 @@ def _transform(
...
+
+    .. v2betastatus:: ToPureTensor transform
+
+    This doesn't scale or change the values, only the type.
+    """
+
+    _transformed_types = (datapoints.Image, datapoints.Video, datapoints.Mask, datapoints.BoundingBoxes)
+
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> torch.Tensor:
+        return inpt.as_subclass(torch.Tensor)
```
Comment (NicolasHug) [inline torchvision/transforms/v2/_type_conversion.py:96]: I don't think we need / want a functional for this. The functional would just wrap `.as_subclass(torch.Tensor)` which is already public and users should just use that

File: torchvision/transforms/v2/_type_conversion.py [hunk 3]
```diff
@@ -80,3 +80,17 @@ def _transform(
...
+
+class ToPureTensor(Transform):
+    """[BETA] Convert all datapoints to pure tensors, removing associated metadata (if any).
+
+    .. v2betastatus:: ToPureTensor transform
+
+    This doesn't scale or change the values, only the type.
+    """
+
+    _transformed_types = (datapoints.Image, datapoints.Video, datapoints.Mask, datapoints.BoundingBoxes)
```
Comment (pmeier) [inline torchvision/transforms/v2/_type_conversion.py:93]: Can't we do

```suggestion
    _transformed_types = (datapoints.Datapoint,)
```

? That would make this transform usable for user defined datapoints as well.

File: test/test_transforms_v2_refactored.py (hunk 1 above)
Comment (NicolasHug) [reply test/test_transforms_v2_refactored.py:2285]: A minor perk of doing it this way is that we also test the "arbitrary structure" support.

File: references/classification/presets.py [hunk 4]
```diff
@@ -68,6 +68,9 @@ def __init__(
         if random_erase_prob > 0:
             transforms.append(T.RandomErasing(p=random_erase_prob))
 
+        if use_v2:
```
Comment (vfdev-5) [inline references/classification/presets.py:71]: technically, we do not need this as we do not wrap to `Image` in the classification pipeline, right ?

Comment (NicolasHug) [reply references/classification/presets.py]: removed

**Pull request merged**